[core]/es_extended/server/onesync.lua

```diff
@@ -15,7 +15,7 @@ local function getNearbyPlayers(source, closest, distance, ignore)
 			error("Received invalid first argument (source); should be playerId or vector3 coordinates")
 		end
 
-		source = GetEntityCoords(GetPlayerPed(source))
+		source = GetEntityCoords(source)
```

Comment:
It is also logically flawed. So if there is no actual source then GetPlayerPed will be nil and if GetPlayerPed is nil then GetEntityCoords will be nil because GetEntityCoords(nil) is received. 

Here you need to specify that if no source is given then the code should not run. 

The other thing is that it is a good thing to overwrite this variable value, just not very transparent. One minute the source is a player ped id, the next minute it's a coordinate. 

My suggestion is to name the variables correctly, we don't need to waste so much memory and it won't happen like what we want to fix. 

So if ped then local playerPedId,if coordinate then local playerCoords or sourcePedId, sourcePedCoords maybe.

Reply:
Yea 👍